### PR TITLE
Add possibility to register with association to existing osbb

### PR DESCRIPTION
**Pull request by Ukhanskyi**
[Trello Task](https://trello.com/c/7eDB9Gy3/66-add-possibility-to-register-with-association-to-existing-osbb)

**CHECK LIST**
- [x] specs added
- [x] specs passed
- [x] code coverage >90%
- [x] DB indexes added if needed
- [ ] seed.rb file is updated if needed
- [ ] PR is reviewed manually again (to make sure you have 100% ready code)
- [x] PR doesn't have conflicts with main branch
- [x] I've checked new feature as logged in and logged out user if needed
- [x] "let!" is used in specs
- [x] screenshot of feature is added if task has UI

image

image

image

### CODE REVIEW

File: app/controllers/user/registrations_controller.rb [hunk 1]
```diff
@@ -1,10 +1,13 @@
 class User::RegistrationsController < Devise::RegistrationsController
   before_action :configure_sign_up_params, only: :create
+  skip_before_action :verify_authenticity_token
```
Comment (vikdotdev) [inline app/controllers/user/registrations_controller.rb:3]: це точно треба?

File: spec/controllers/search_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+    it 'returns success when searchs osbb' do
+      Osbb.search(osbb.name)
+      expect(response).to be_successful
+    end
+
+    it 'finds a searched osbb by name' do
+      result = Osbb.search(osbb.name)
+      expect(result).to match([osbb])
+    end
```
Comment (vikdotdev) [inline spec/controllers/search_controller_spec.rb:25]: в контроллер тесті тестуєм контроллер, а не код взятий з контроллера

Comment (Ukhanskyi) [reply spec/controllers/search_controller_spec.rb:25]: Вже покрив тестами той контроллер)

File: app/controllers/user/registrations_controller.rb (hunk 1 above)
Comment (Ukhanskyi) [reply app/controllers/user/registrations_controller.rb]: Вже не потрібно

File: app/controllers/search_controller.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class SearchController < ApplicationController
+  def search
```
Comment (chu4) [inline app/controllers/search_controller.rb:2]: specify what kind of search it is
to understand what records info will be returned

File: spec/controllers/registrations_controller_spec.rb [hunk 4]
```diff
@@ -7,6 +7,8 @@
                 .merge({ password: '123456', confirm_password: '123456' })
     end
     let!(:invalid_attributes_user) { FactoryBot.build(:user).attributes }
+    let!(:osbb) { create(:osbb, id: 13) }
```
Comment (igorkasyanchuk) [inline spec/controllers/registrations_controller_spec.rb:10]: а чого 13? це що за прикол

File: spec/controllers/search_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,73 @@
+require 'rails_helper'
+
+RSpec.describe SearchController, type: :controller do
+  let!(:osbb) do
```
Comment (igorkasyanchuk) [inline spec/controllers/search_controller_spec.rb:4]: Додай таке в тестах є render_views для контролерів

**Review by matsevich — APPROVED** [review]

File: spec/controllers/registrations_controller_spec.rb (hunk 4 above)
Comment (Ukhanskyi) [reply spec/controllers/registrations_controller_spec.rb]: виправив

File: spec/controllers/search_controller_spec.rb (hunk 5 above)
Comment (Ukhanskyi) [reply spec/controllers/search_controller_spec.rb]: готово

File: app/controllers/search_controller.rb (hunk 3 above)
Comment (Ukhanskyi) [reply app/controllers/search_controller.rb]: It seems I have corrected.

**Comment by Ukhanskyi** [comment]
image

**Review by IvanDanyliv — APPROVED** [review]

**Review by chu4 — APPROVED** [review]

**Pull request merged**